src/js/tracks/track.js

```diff
@@ -91,6 +83,33 @@ class Track extends EventTarget {
         set() {}
       });
     }
+
+    /**
+     * @memberof Track
+     * @member {string} label
+     *         The label of this track. Cannot be changed after creation.
+     * @instance
+     *
+     * @fires Track#labelchange
+     */
+    Object.defineProperty(this, 'label', {
+      get() {
+        return label;
+      },
+      set(newLabel) {
+        label = newLabel;
+
+        /**
+         * An event that fires when label changes on this track.
+         *
+         * > Note: This is not part of the spec!
+         *
+         * @event Track#labelchange
+         * @type {EventTarget~Event}
+         */
+        this.trigger('labelchange');
```

Comment:
Should we only trigger this label actually changed? ie:
```js
if (label !== newLabel) { 
  label = newLabel;
  this.trigger('labelchange')
}
```

Reply:
Yeah, this makes sense.

Reply:
Good call- added a check for this!